posthog/temporal/batch_exports/spmc.py

```diff
@@ -1244,20 +1251,20 @@ async def start(
                     bytes_count / 1024**2,
                 )
 
-                # Transform batch to chunks
-                # We also split the file if we reach the max file size
-                for chunk in transformer.transform_batch(record_batch):
-                    await self.consume_chunk(data=chunk)
-                    current_file_size += len(chunk)
-                    if max_file_size_bytes and current_file_size > max_file_size_bytes:
-                        for chunk in transformer.finalize():
-                            await self.consume_chunk(chunk)
-                        await self.finalize_file()
-                        current_file_size = 0
-
-            # Finalize transformer and consume any remaining data
-            for chunk in transformer.finalize():
-                await self.consume_chunk(chunk)
+                yield record_batch
+
+        current_file_size = 0
+        try:
+            async for chunk in transformer.iter_transformed_record_batches(
+                track_iteration_of_record_batches(),
+                max_file_size_bytes,
+            ):
+                await self.consume_chunk(data=chunk)
+                current_file_size += len(chunk)
+
+                if max_file_size_bytes and current_file_size > max_file_size_bytes:
```

Comment:
We might run into issues tracking the current file size both here and inside the transformer. For example, the `transformer` can yield extra chunks of data when finalizing the file, which would be included in the `current_file_size` here. It's probably not an issue since we'll still call `self.finalize_file` here but feels like it would be better to keep track of the file size in a single place

Reply:
Parquet shouldn't be running this. But I'll review.

Reply:
I agree, I think we can probably have the transformer return a sentinel when it reaches EOF.